monai/apps/pathology/transforms/spatial/dictionary.py

```diff
@@ -53,4 +54,80 @@ def __call__(self, data: Mapping[Hashable, torch.Tensor]) -> Dict[Hashable, torc
         return d
 
 
+class TileOnGridd(MapTransform):
+    """
+    Tile the image into patches on a grid and maintain a subset of it.
+    This transform works only with np.ndarray inputs.
+
+    Args:
+        tile_count: number of tiles to extract
+        tile_size: size of the square tile
+            Defaults to ``256``.
+        tile_step: step size
+            Defaults to None (same as tile_size)
+        tile_all: Extract all non-background tiles, instead of tile_count.
+            Defaults to ``False``.
+        tile_random_offset: Randomize position of tile grid, instead of starting from the top-left corner
+            Defaults to ``False``.
+        tile_pad_full: pad image to the size evenly divisible by tile_size
+            Defaults to ``False``.
+        tile_background_val: the background constant (e.g. 255 for white background)
+            Defaults to ``255``.
+        tile_filter_mode: [min, max, None] filter tiles. If total number of tiles is more then tile_size,
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for None) subset
+            Defaults to ``min`` (which assumes background is white, high value)
+        random_seed: provide random seed int manually
+            Defaults to ``None``
+        return_list_of_dicts: return each tile in a separate dictionary, as a list of dicts
+            Defaults to ``False``
+
+    """
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        tile_count: int = None,
+        tile_size: int = 256,
+        tile_step: int = None,
+        tile_all: bool = False,
+        tile_random_offset: bool = False,
+        tile_pad_full: bool = False,
+        tile_background_val: int = 255,
+        tile_filter_mode: str = "min",
+        allow_missing_keys: bool = False,
+        random_seed: int = None,
+        return_list_of_dicts: bool = False,
+    ):
+        super().__init__(keys, allow_missing_keys)
+
+        self.return_list_of_dicts = return_list_of_dicts
+        if return_list_of_dicts and len(keys) > 1:
+            raise ValueError("Only a single key is supported when returning a list of dicts.")
+
+        self.splitter = TileOnGrid(
+            tile_count=tile_count,
+            tile_size=tile_size,
+            tile_step=tile_step,
+            tile_all=tile_all,
+            tile_random_offset=tile_random_offset,
+            tile_pad_full=tile_pad_full,
+            tile_background_val=tile_background_val,
+            tile_filter_mode=tile_filter_mode,
+            random_seed=random_seed,
+        )
+
+    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+
+        d = dict(data)
+        for key in self.key_iterator(d):
+            d[key] = self.splitter(d[key])
+
+            if self.return_list_of_dicts:
+                image = d.pop(key, None)  # remove this key
+                d = [{**d, **{key: image[i]}} for i in range(len(image))]
```

Comment:
I feel there are 2 issues here:
1. Should use `deepcopy` to copy other things into every item of the list.
2. If I have multiple keys, this line seems can't generate list of dicts as expected?

I think you can refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/dictionary.py#L1160

Thanks.

Reply:
1. we don't need deep copy - because other keys will be used only from the first Dict later (and all the other dismissed) in collate function.  you sure you want deepcopy?
2. Multiple keys for Tiling this way are not supported, only a single key will be supported.  (that's the plan).  If we try to support multiple keys, each key can potentially have a different number of tiles in nd.array.  So it's not clear how to split them into a single list of dicts

Reply:
Hi @myron ,

1. In your current tutorial use case, it doesn't need to `deepcopy` the other keys. But if other users try to apply this new transform in other tasks, I think the shallow-copy may cause unknown errors.
2. If it truly can only support 1 key here, please add some check for that: if `allow_missing_keys is False` and keys > 1, raise exception to nofity users it only supports 1 key. And no need to execute for-loop for keys anymore.

Thanks.

Reply:
ok, I've added deepcopy, and made it work for any number of keys